G0 slider 49 1.0

ducky64 opened the pull request:
It seems to build, announce on the JACDAC network, and show up on the dashboard.

## TODOs
Because parts are delayed, have not tested the LEDs or the slider itself. But touching the pot pin does cause the slider value to change, so it's definitely doing something.

- [x] Test LEDs
- [x] Test with slider populated
- [ ] Maybe calibrate the RGBs ... how?

ducky64 commented:
The slider reads from the dashboard and all the LEDs seem to work (by recompiling with different pins mapped). Red one is a bit dim and green one is a bit bright, but probably an artifact of the device itself. The LED brightness constants are not calibrated, not sure how we do that...

jamesadevine commented:
> The LED brightness constants are not calibrated, not sure how we do that...

I think @mmoskal usually uses his lux meter... :smile: 

Which RGB LED did you fit in the end?

jamesadevine commented on targets/jm-slider-49-1.0/board.h:
```diff
@@ -0,0 +1,27 @@
+#define PIN_BL_PERIOD 300
+
+// TODO WHAT IS THIS
+#define BOARD_STARTUP_CODE LL_SYSCFG_EnablePinRemap(LL_SYSCFG_PIN_RMP_PA12);
```
This is a G0 thing... some of the pins are dual function.

![image shows STM datasheet, PA12[PA10] is in focus.](https://user-images.githubusercontent.com/4742697/124725646-ec2a7200-df04-11eb-8fb8-81d478502f9a.png)

This line will mean that instead of PA12 having the functionality of PA10, it will instead internally remap to PA10 and have its functionality:
![STM32g030 datasheet, PA10 and its functionality are highlighted](https://user-images.githubusercontent.com/4742697/124725859-2136c480-df05-11eb-9e47-42d08263737a.png)

ducky64 replied:
Since I don't use that pin, I assume it's safe to delete that line?

mmoskal reviewed (approved):
but please remove the TODO comment!

ducky64 commented:
Most of the boards have the inexpensive LED that I don't think we've used before:
PART: 3147-B38G3RGB-10D0003H2U1930CT-ND
MFG : Harvatek Corporation (VA) / B38G3RGB-10D0003H2U1930 

1-2 boards have the LED specified on prior boards (as a reference and hedge), but the price has now gone through the roof:
PART: 516-1795-1-ND 
MFG : Broadcom Limited (VA) / HSMF-C114

jamesadevine commented:
I’d assume so, but test it on hardware to be sure!

James
________________________________
From: Richard Lin ***@***.***>
Sent: Wednesday, July 7, 2021 6:45:02 PM
To: microsoft/jacdac-msr-modules ***@***.***>
Cc: James Devine ***@***.***>; Review requested ***@***.***>
Subject: Re: [microsoft/jacdac-msr-modules] G0 slider 49 1.0 (#17)


@ducky64 commented on this pull request.

________________________________

In targets/jm-slider-49-1.0/board.h<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fjacdac-msr-modules%2Fpull%2F17%23discussion_r665583774&data=04%7C01%7Ct-jamdev%40microsoft.com%7Cc89fd3d36ac44518037208d9416ef334%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637612767045652274%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=HBjaO71G8k%2B1ZqdXIJcfBySpG7NjsFHYunbtYGbFiS4%3D&reserved=0>:

> +#define PIN_LED_R PB_0
+#define PIN_LED_G PA_7
+#define PIN_LED_B PA_6
+
+// calibrated to ~50lux per channel
+// TODO needs calibration
+#define LED_R_MULT 250
+#define LED_G_MULT 150
+#define LED_B_MULT 42
+#define RGB_LED_PERIOD 600
+
+#define PIN_BL_LED PIN_LED_B
+#define PIN_BL_PERIOD 300
+
+// TODO WHAT IS THIS
+#define BOARD_STARTUP_CODE LL_SYSCFG_EnablePinRemap(LL_SYSCFG_PIN_RMP_PA12);


Since I don't use that pin, I assume it's safe to delete that line?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fjacdac-msr-modules%2Fpull%2F17%23discussion_r665583774&data=04%7C01%7Ct-jamdev%40microsoft.com%7Cc89fd3d36ac44518037208d9416ef334%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637612767045662261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=rTpDR8a3t5l23aa6NvmO3lcBzq4RKrVL5oblRA6NrCI%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEF4KIC554BEFW6FY54KFTTWSHB5ANCNFSM47TKUSQQ&data=04%7C01%7Ct-jamdev%40microsoft.com%7Cc89fd3d36ac44518037208d9416ef334%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637612767045662261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=3sLWDrdcOpTuKMqdcwqtW2jCSNhq%2BtduX8awQqwUF90%3D&reserved=0>.

ducky64 commented:
Removed the TODOs and the unnecessary PA10/12 config, and added some comments on the calibration ('taken from this other board with this P/N and resistors'). Will merge if no more comments.

jamesadevine commented:
Thanks Richard!
